src/Map/Sketch/useSketch.ts

```diff
@@ -0,0 +1,558 @@
+import { LineString, MultiPolygon, Point, Polygon, Feature } from "geojson";
+import { Dispatch, MutableRefObject, SetStateAction, useCallback, useEffect, useRef } from "react";
+import invariant from "tiny-invariant";
+import { RefObject } from "use-callback-ref/dist/es5/types";
+
+import { Position3d } from "../../types";
+import { EngineRef, MouseEventCallback, MouseEventProps } from "../types";
+
+import { SketchInterpreter, sketchState } from "./hooks";
+import {
+  GeometryOptionsXYZ,
+  SketchEditingFeature,
+  SketchFeature,
+  SketchOptions,
+  SketchType,
+} from "./types";
+
+type Props = {
+  state: sketchState;
+  send: SketchInterpreter["send"];
+  engineRef: RefObject<EngineRef>;
+  disableInteraction: boolean;
+  type: SketchType | undefined;
+  updateType: Dispatch<SetStateAction<SketchType | undefined>>;
+  geometryOptions: GeometryOptionsXYZ | null;
+  setGeometryOptions: Dispatch<SetStateAction<GeometryOptionsXYZ | null>>;
+  updateGeometryOptions: (controlPoint?: Position3d) => void;
+  sketchEditingFeature?: SketchEditingFeature;
+  setSelectedControlPointIndex: Dispatch<SetStateAction<number | undefined>>;
+  markerGeometryRef: MutableRefObject<GeometryOptionsXYZ | null>;
+  pointerLocationRef: MutableRefObject<[lng: number, lat: number, height: number] | undefined>;
+  extrudedHeight: number;
+  setExtrudedHeight: (height: number) => void;
+  setExtrudedPoint: (point: Position3d | undefined) => void;
+  updateCentroidPoints: (controlPoints: Position3d[]) => void;
+  createFeature: () => Feature<
+    Polygon | MultiPolygon | Point | LineString,
+    {
+      id: string;
+      type: SketchType;
+      positions: Position3d[];
+      extrudedHeight: number;
+    }
+  > | null;
+  handleFeatureCreate: (feature: SketchFeature) => void;
+  applyEdit: () => void;
+  cancelEdit: () => void;
+  isEditing: boolean;
+  sketchOptions: SketchOptions;
+};
+
+const movingStatus = [
```

Comment:
Could you add a type to this variable? It's difficult to notice a change when the name is changed or removed.

Reply:
Updated!